src/infra/AutocompleteServiceImpl.ts

```diff
@@ -70,6 +70,7 @@ export class AutocompleteServiceImpl implements AutocompleteService {
       version: this.config.experienceVersion,
       locale: this.config.locale,
       sessionTrackingEnabled: request.sessionTrackingEnabled,
+      visitor: JSON.stringify(this.config.visitor || undefined),
```

Comment:
if this.config.visitor is not defined, it would automatically be JSON.stringify(undefined) so I think we can remove `|| undefined`